src/components/Nav/Nav.jsx

```diff
@@ -1,18 +1,68 @@
-import React from 'react';
-import { useNavigate } from 'react-router-dom';
+import React, { useState, useEffect } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { APIS } from '../../config.js';
+import { KAKAO_AUTH_URL } from '../SocialOAuth.js';
 import { S } from './Nav.js';
 const Nav = () => {
+  const [isSignedIn, setIsSignedIn] = useState(false);
   const navigate = useNavigate();
+  const location = useLocation();
+  const accessToken = localStorage.getItem('accessToken');
+
+  const handleSignIn = () => {
+    window.location.href = `${KAKAO_AUTH_URL}`;
+  };
+
+  const handleSignOut = () => {
+    const accessToken = localStorage.getItem('accessToken');
+
+    fetch(`${APIS.users}/kakao/signout`, {
+      method: 'POST',
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+        'Content-Type': 'application/json;charset=utf-8',
+      },
+    })
+      .then(res => res.json())
+      .then(data => {
+        if (data.message === 'Logout successful') {
+          localStorage.removeItem('accessToken');
+          alert('로그아웃 되었습니다!');
+          navigate('/');
+        } else throw new Error('로그아웃 에러');
+      });
+  };
+
+  useEffect(() => {
+    accessToken ? setIsSignedIn(true) : setIsSignedIn(false);
+  }, [location.pathname, accessToken]);
 
   return (
     <S.NavContainer>
       <div>
         <ul>
           <li>
             <S.NavTopWrap>
-              <S.LoginBtn>LOGIN</S.LoginBtn>
+              {!isSignedIn ? (
+                <S.LoginBtn onClick={() => handleSignIn()}>LOGIN</S.LoginBtn>
+              ) : (
+                <>
+                  <S.LogoutBtn onClick={() => handleSignOut()}>
```

Comment:
```suggestion
                <S.LoginBtn onClick={handleSignIn}>LOGIN</S.LoginBtn>
              ) : (
                <>
                  <S.LogoutBtn onClick={handleSignOut}>
```
함수에 인자로 전달하는 값이 없을때는 위와 같이 작성하는 것이 더 효율적입니다.